Create 24090 Release Notes.md (#297)
o3de/sig-release: releases/24.09.0/24090 Release Notes.md

See Simulation, Rendering, and AR/VR sections for specific changes in these areas.

## General Changes and Improvements in 24.09.0

**Major Improvements:**

* Quick-Start “Script Only” Projects \- Can build projects using Script Canvas and Lua, without needing a C++ compiler,  using the O3DE installer version.  
* Many optimizations for memory usage on mobile and AR/VR devices  
* Mobile performance \- up to 400% improvement across iOS, Android and AR/VR/XR devices  
* Split PhysX 4 and 5 into separate gems, enabling users to switch between the PhysX versions easily, even when using installer builds.  
* Added a Project Export UI along wth full export support for iOS, Android, Linux, and Windows. Users are no longer required to use a command line.  
* Added options to reduce size of headless server up to 90%, dependent on assets used.   
* Added a mobile specific rendering pipeline, allowing users to easily enable/disable features as needed.  
* Added ability to control quality settings per device. There is a default of 3 performance levels (low, medium, high) based on device specifications for CPU, GPU, and memory.  

> * Added shader variants. When setup, this allows the renderer to automatically utilize the most performant shader for a given rendering need.   

Reviewer: I think this needs rewording, because as described, this feature has been available for a long time now? If this trying to describe Specialization Constants? or improvements in the ShaderManagementConsole?

* Resolved many Script Canvas quality of life issues  
* Improved editor performance with large number of entities  
* Added support for ray-traced reflections in SpecularReflectionsFeatureProcessor  
* Many Asset Browser quality of life improvements  
* Terrain | Add Help Url to components that have docs pages   
* MiniAudio: Added more volume controls and a pause method for Playback component  
* Add quality settings to default project template  
* Many Atom performance improvements for all supported platforms  
* Added support for renaming gems  
* Added Support For Raytracing Against Skinned Meshes  
* Added functionality for material pipeline files to specify custom object SRG members  
* Added support for lighting channels to all light types  
* Added support for configuring anti-aliasing and multi-sample count via CVars  
* Added support of node-searching in script canvas  
* Added support to turn on/off directional lights shadow  
* Added support to run clients headlessly for simulation client  
* MultiplayerSample gamepad player input support  
* Added vertex color support to BasePBR, StandardPBR, EnhancedPBR material types as well as material canvas.  

> * [Added three new tone-mapping modes: AcesFitted, AcesFilmic and Filmic](https://github.com/o3de/o3de/commit/af81e8362e5a9a15047d35d34573e4aacceb7941)  
> * [Added shadow support for SimpleSpotLights](https://github.com/o3de/o3de/commit/7ab669f974a6df543339a1cc625c34f6ad21b282)  
> * [Added Gobo (i.e texture projection) support for simple spot light.](https://github.com/o3de/o3de/commit/08e1d396a40b56db24ee26952da935c625d9def1)  

Reviewer 2: I would remove the links (or add them everywhere) to keep the list more consistent
```suggestion
* Added three new tone-mapping modes: AcesFitted, AcesFilmic and Filmic
* Added shadow support for SimpleSpotLights
* Added Gobo (i.e texture projection) support for simple spot light.
```

* ROS 2 Gem: modified clock designed  
* ROS 2 RobotImporter: added Gazebo data parsing for topics, frames and namespaces  
* ROS 2 RobotImporter: redesigned wizard

**Fixes**

* ROS 2 Gem: fixed ROS2SpawnerComponent  
* ROS 2 Gem: fixed Lidar data publishing  
* ROS 2 related Gems and project templates: cleaned-up versioning, updated to work with ROS 2 Jazzy  
* ROS 2 related Gems and project templates: updated prefabs to match O3DE \>= 2310.0

**Removed code**

* ROS 2 Gem: removed outdated and duplicated documentation files from the Gem

**Other Gems worth noticing:**

* Pointcloud https://github.com/RobotecAI/robotec-o3de-tools/tree/main/Gems/Pointcloud  
* ROS2ScriptIntegration https://github.com/RobotecAI/robotec-o3de-tools/tree/main/Gems/ROS2ScriptIntegration  
* VehicleDynamics https://github.com/RobotecAI/o3de-vehicle-dynamics-gem

## Rendering Specific Changes for 24.09.0 Release

**Mobile Render Pipeline \-** Add support for an extremely performant mobile render pipeline built for tiled GPUs.

* Optimised BRDF for mobile \- Added a new optimised BRDF for mobile as well as support for analytical Env IBL approximation.  

> * [Added Gobo (i.e texture projection) support for simple spot light.](https://github.com/o3de/o3de/commit/08e1d396a40b56db24ee26952da935c625d9def1)  
> * [Added CPU culling for decals](https://github.com/o3de/o3de/commit/6d23563d2b0633409969aeff29f9d10f12c6c875).  
> * [Added shadow support for SimpleSpotLights](https://github.com/o3de/o3de/commit/7ab669f974a6df543339a1cc625c34f6ad21b282).  
> * [Added support for decal colour and decal colour factor to help modulate final calculated decal colour.](https://github.com/o3de/o3de/commit/ed2f49bd2d56137e0ef88c9094dda1f328a8f401)  
> * [Added shader options for simple point/spot lights and decals](https://github.com/o3de/o3de/commit/c8ca92868bff7170bbf4e5d36154e7ed761bd89a).   
> * [Added support for cpu culling for simple-point and simple-punctual lights](https://github.com/o3de/o3de/commit/b6800d426a9e074c9f8a08dc6f9cdaf251ee95ca).  

Reviewer 2: I would suggest removing the links to keep it consistent; I would suggest passing through a spellcheck to unify the language (`colour` -> `color`) in this example
```suggestion
* Added Gobo (i.e texture projection) support for simple spot light.
* Added CPU culling for decals
* Added shadow support for SimpleSpotLights
* Added support for decal color and decal color factor to help modulate final calculated decal color.
* Added shader options for simple point/spot lights and decals
* Added support for cpu culling for simple-point and simple-punctual lights
```

* Added support for deferred fog within the mobile pipeline.  
* Added exponential and exponential squared fog modes.   

> * [Added support for shader options within PCF filtering for directional light shadows](https://github.com/o3de/o3de/commit/75bc7b72446a34f351929e1d917b25427725a5c2).  

Reviewer 2: ```suggestion
* Added support for shader options within PCF filtering for directional light shadows
```

* Added Silhouette support to the mobile render pipeline. For more details about the Silhouette feature see the section about Rendering additions.  
* Added particles support to the mobile render pipeline.  
* Added support for postFX effects within the pipeline  
* Various half float optimizations for the mobile render pipeline.  

> * [Render scale related support and fixes](https://github.com/o3de/o3de/commit/64edaebaf2c34bb99c733c6c013c1614ba95dfdf) across various mobile devices.

Reviewer 2: ```suggestion
* Render scale related support and fixes across various mobile devices.
```

**Rendering additions/improvements \-** Various rendering improvements for all the render pipelines and general Atom improvements.

* Initial scaffolding for Multi-GPU support within RHI and RPI layers of the engine. A lot of this work includes initial plumbing for Multi-GPU objects and initial support for basic GPU-GPU synchronisation when transferring resources between devices.  
* Silhouette Feature \- Add support for Silhouette material type that, when applied to a mesh, draws an outline and filled-in silhouette in various modes: Always, XRay, Visible and Never.  
* Added vertex colour support  to BasePBR, StandardPBR, EnhancedPBR material types as well as material canvas.  

> * [Added three new tone-mapping modes: AcesFitted, AcesFilmic and Filmic](https://github.com/o3de/o3de/commit/af81e8362e5a9a15047d35d34573e4aacceb7941). Mobile render pipeline defaults to Filmic mode.  

Reviewer 2: ```suggestion
* Added three new tone-mapping modes: AcesFitted, AcesFilmic and Filmic. Mobile render pipeline defaults to Filmic mode.  
```

* Added android and ios registry settings for four tiers (low, medium, high, very high) of graphics quality.  
* Added support for geometry shader within DX12 and Vulkan RHI backends.  

> * [Added support for intersection shaders](https://github.com/o3de/o3de/commit/820e6b3c30136ed9212aae26803b2ceeddba16bb) within DX12 and Vulkan RHI backends.  

Reviewer 2: ```suggestion
* Added support for intersection shaders within DX12 and Vulkan RHI backends.
```

* r\_antiAliasing \- Add support to switch anti-aliasing modes via CVar r\_antiAliasing  
* Add support for lighting channels \- This feature introduces 3 channels for each kind of light and light-able objects. The objects and lights in the same channel are grouped and as a result the lights in the group affect the objects in the same group. The default behaviour is all lights and objects in channel 0, so it should not change the current way to work with lighting, but it will provide a further option to grant the artists the flexibility to control which light shines on which objects.  
* Enable and disable ContrastAdaptiveSharpening pass as well along with temporal anti aliasing (i.e TAA).  
* Added support to allow passing customised derivatives to texture sampling functions.  
* Added device attribute for the GPU model. This allows us to provide better device filtering for graphics quality tiers.  
* Added per mesh shader option support when mesh instancing is enabled. This allows us to build optimised shader variants for a scene with mesh instancing.  
* Added support for timeline semaphores within Vulkan RHI to help with low level synchronisation.  

> * [Added ability to disable creating a native window for game launchers](https://github.com/o3de/o3de/commit/66f0ea88fff686f43f1ec7b1e2b717d0b2eb8631) for null renderers.

Reviewer 2: ```suggestion
* Added ability to disable creating a native window for game launchers for null renderers.
```

**Atom (O3DE renderer) Memory optimizations**

* Added support to dump information about loaded assets.  
* Various Metal RHI related memory optimizations.  
* Added support to release BufferAssets owned by ModelAssets. This allows us to release mesh related memory on the cpu once it is transferred to the gpu.  

> * [Ring buffer support for RayTracing buffers](https://github.com/o3de/o3de/commit/c7c6817c9a374ddbce33eb8c4563335a06b82477).

Reviewer 2: ```suggestion
* Ring buffer support for RayTracing buffers.
```

Martin Winter  
lemonade-dm  
guillaume-haerinck  
Akio Gaule  
Junhao Wang  
Danilo Aimini  
Shauna Gordon  
Joe Bryant  
Markus Prettner  
Martin Winter  
colinb\[APMG\]  
antonmic  
Joerg H. Mueller  
antonmic  
Hongshan Li  
Artur Kamieniecki  
siliconvoodoo  
Martin Sattlecker  
Chris Galvan  
Qing Tao  
Pawel Liberadzki  
Nicholas Lawson  
pwalterscarb  
Sarah Anderson  
André L. Alvares  
Reece Hagan  
Adi Bar-Lev  
omacx  
Olex Lozitskiy  
phuchau1989  
Patryk  
Xichen Zhou  
kursad-k  
Paulina Kubera  
carbonated-niko  
stevo26  
Djfos  
TheTopHat26  
Mateusz Żak  
Naomi Washington  
Vincent  
Antoni-Robotec  
0x14307  
Kacper Lewandowski  
o3de-issues-bot  
Patryk Antosz  
ShalokShalom  
jjs-home  
ExMatics HydrogenC  
realJavabot  
Nathan Brooks  
idclosed  
rhuthCarb  
AaronRuizMoraUK  
Umesh Rajesh Ramchandani  
Guy MacGeorges  
Alexander  
Lloyd Tullues  
Huawei-CarlosCarbone  
Shao Biyao  
Antoni Puch  
Andre Mitchell  
solo1967  
LB-MichalRodzos  
Luis Gutierrez  
Wiktor Bajor  
Mateusz Żak  
Maciej Aleksandrowicz  
CarbAndrewD  
Roderick Kieley  
konkuk kim  
Caleb Leak  
TheShed412  
Fergal  
utk-hua  
Tobias Alexander Franke  
Yaakuro  
Adam Krawczyk  

> knicked  

Reviewer 2: The same author, `Kacper Dąbrowski` is mentioned in line 365
```suggestion
```
